src/behavior/impl/balancer-jar.ts

```diff
@@ -1,27 +1,108 @@
 import { Provider, TransactionResponse } from "@ethersproject/providers";
 import {
   BigNumber,
+  Contract,
   ContractTransaction,
   ethers,
   Signer,
 } from "ethers";
 import strategyAbi from "../../Contracts/ABIs/strategy.json";
-import { PickleModel } from "../..";
-import { HistoricalYield, JarDefinition } from "../../model/PickleModelJson";
-import { getBalancerPerformance } from "../../protocols/BalancerUtil";
+import jarAbi from "../../Contracts/ABIs/jar.json";
+import { JarHarvestStats, PickleModel } from "../..";
+import {
+  AssetProjectedApr,
+  HistoricalYield,
+  JarDefinition,
+} from "../../model/PickleModelJson";
+import {
+  calculateBalPoolAPRs,
+  getBalancerPerformance,
+  getPoolData,
+  PoolData,
+} from "../../protocols/BalancerUtil";
 import { BalancerClaimsManager } from "../../protocols/BalancerUtil/BalancerClaimsManager";
 import { AbstractJarBehavior } from "../AbstractJarBehavior";
 import { Prices } from "../../protocols/BalancerUtil/types";
 import { ICustomHarvester, PfCoreGasFlags } from "../JarBehaviorResolver";
 
-export abstract class BalancerJar extends AbstractJarBehavior {
+export class BalancerJar extends AbstractJarBehavior {
+  poolData: PoolData | undefined;
+
+  async getDepositTokenPrice(
+    jar: JarDefinition,
+    model: PickleModel,
+  ): Promise<number> {
+    if (!this.poolData) {
+      try {
+        this.poolData = await getPoolData(jar, model);
+      } catch (error) {
+        const msg = `Error in getDepositTokenPrice (${jar.details.apiKey}): ${error}`;
+        console.log(msg);
+        return 0;
+      }
+    }
+
+    return this.poolData.pricePerToken;
+  }
+
+  async getProjectedAprStats(
+    jar: JarDefinition,
+    model: PickleModel,
+  ): Promise<AssetProjectedApr> {
+    if (!this.poolData) this.poolData = await getPoolData(jar, model);
+    const res = await calculateBalPoolAPRs(jar, model, this.poolData);
+    const aprsPostFee = res.map((component) =>
+      this.createAprComponent(
+        component.name,
+        component.apr,
+        component.compoundable,
+      ),
+    );
+    return this.aprComponentsToProjectedApr(aprsPostFee);
+  }
+
   async getProtocolApy(
     definition: JarDefinition,
     model: PickleModel,
   ): Promise<HistoricalYield> {
     return await getBalancerPerformance(definition, model);
   }
 
+  async getAssetHarvestData(
+    definition: JarDefinition,
+    model: PickleModel,
+    balance: BigNumber, // total want balance in strategy+jar
+    available: BigNumber, // strategy want balance + jar earnable balance (95% of jar want balance)
+    resolver: Signer | Provider,
+  ): Promise<JarHarvestStats> {
+    const ret = await super.getAssetHarvestData(
```

Comment:
I don't see a reason to override the default implementation. The default `earnableUSD` would yield the same number here.